src/com/owncloud/android/datamodel/OCFile.java

```diff
@@ -747,4 +749,36 @@ public void copyLocalPropertiesFrom(OCFile sourceFile) {
         setEtagInConflict(sourceFile.getEtagInConflict());
     }
 
+    /**
+     * Manipulate file remote id to get private link
+     *
+     * @param context
+     * @param account
+     * @return private link for the current file
+     */
+    public String getPrivateLink(Context context, Account account) {
+
+        // Parse remoteId
+        String parsedRemoteId = mRemoteId.substring(0, Math.min(mRemoteId.length(), 8));
```

Comment:
Sorry, I don't understand what conversion are you doing here to get the string for the private link